packages/buendia-networking/data/usr/share/buendia/config.d/10-networking

```diff
@@ -12,192 +12,223 @@
+        iface lo inet loopback
+    " > /etc/network/interfaces
+}
+
+# Activate or deactivate the DHCP and DNS servers according to the settings
+configure_services () {
+    if bool $NETWORKING_DHCP_DNS_SERVER; then
+        # If a DHCP range is set, then assume we want to be a DHCP server as
+        # well as a DNS server.
+        if [ -n "$NETWORKING_DHCP_RANGE" ]; then
+            echo 'Turning on DHCP and DNS server.'
+            ntp_address=$NETWORKING_IP_ADDRESS
+            if [ -z "$ntp_server" -a -n "$NETWORKING_WIFI_ADDRESS" ]; then
+                ntp_address=$NETWORKING_WIFI_ADDRESS
+            fi
+            if [ -z "$ntp_server" -a -n "$NETWORKING_ETHERNET_ADDRESS" ]; then
+                ntp_address=$NETWORKING_ETHERNET_ADDRESS
+            fi
+            unindent <<< "
+                dhcp-authoritative
+                dhcp-range=$NETWORKING_DHCP_RANGE
+                dhcp-option=option:ntp-server,$ntp_address
+            " > /etc/dnsmasq.d/buendia-dhcp.conf
+        else
+            # Otherwise, serve DNS but not DHCP.
+            echo "Turning on DNS server."
+            rm -f /etc/dnsmasq.d/buendia-dhcp.conf
+        fi
+
+        # Call update-hosts whenever an interface comes up
+        ln -sf /usr/bin/buendia-update-hosts /etc/network/if-up.d/
+                
+        service dnsmasq restart
+        update-rc.d dnsmasq enable
     else
-        echo "Turning on DNS server."
-        rm -f /etc/dnsmasq.d/buendia-dhcp.conf
-    fi
+        echo "Turning off DHCP/DNS server."
+        rm -f /etc/network/if-up.d/buendia-update-hosts
 
-    ln -sf /usr/bin/buendia-update-hosts /etc/network/if-up.d/
-            
-    service dnsmasq restart
-    update-rc.d dnsmasq enable
-else
-    echo "Turning off DHCP/DNS server."
-    rm -f /etc/network/if-up.d/buendia-update-hosts
-
-    # Disable dnsmasq now, and also for future boots.
-    service dnsmasq stop
-    update-rc.d dnsmasq disable
-fi
+        # Disable dnsmasq now, and also for future boots.
+        service dnsmasq stop
+        update-rc.d dnsmasq disable
+    fi
+}
 
-# Activate or deactivate access point mode according to the settings.
-if bool $NETWORKING_AP; then
-    echo "Becoming an AP, creating wifi network '$NETWORKING_SSID'."
-    hostapd_conf=/etc/hostapd/hostapd.conf
-    unindent <<< "
-        interface=$NETWORKING_WIFI_INTERFACE
-        ssid=$NETWORKING_SSID
-        wpa=2
-        wpa_key_mgmt=WPA-PSK
-        wpa_passphrase=$NETWORKING_PASSWORD
-        hw_mode=g
-        channel=24
-        ieee80211n=1
-        wmm_enabled=1
-        macaddr_acl=0
-        auth_algs=1
-        ignore_broadcast_ssid=0
-        rsn_pairwise=CCMP
-    " > $hostapd_conf
-
-    # Turn this machine into an access point now, and also on future boots.
-    systemctl enable hostapd
-    systemctl start hostapd
-
-else
-    echo "Joining existing wifi network '$NETWORKING_SSID'."
+# Create a file that records the requested settings in a canonical format.
```

Comment:
This comment talks about creating a file, but the file isn't created here.  I think it makes sense to factor out a separate function just for all the tmp_settings stuff (lines 135 to 153).